### PR TITLE
OSD panic due to directory split (bsc#1183760)

### PR DESCRIPTION
**Pull request by tbazant**

### CODE REVIEW

**Review by trociny — COMMENTED** [review]
LGTM.

One thing: I would mention that it is relevant only for filestore. If bluestore is used (which is default and recommended) naturally there is no any directory split.

File: xml/admin_ceph_troubleshooting.xml [hunk 1]
```diff
@@ -518,4 +518,34 @@ keep_jobs: 1</screen>
    </procedure>
   </tip>
  </sect1>
+ <sect1 xml:id="trouble-osd-panic">
+  <title>OSD Panic Occurs when Media Error Happens during Directory Split</title>
```
Comment (trociny) [inline xml/admin_ceph_troubleshooting.xml:522]: Probably: s/OSD Panic Occurs when Media Error Happens during Directory Split/OSD Panic Occurs when Media Error Happens during Filestore Directory Split/

File: xml/admin_ceph_troubleshooting.xml [hunk 2]
```diff
@@ -522,7 +522,7 @@ keep_jobs: 1</screen>
   <title>OSD Panic Occurs when Media Error Happens during Directory Split</title>
 
   <para>
-   When directory split error is generated, the corresponding OSD is designed
+   When directory split error is generated on &filestore; OSDs, the corresponding OSD is designed
```
Comment (trociny) [inline xml/admin_ceph_troubleshooting.xml]: :+1:  I still would like the filestore is mentioned in the title.

File: xml/admin_ceph_troubleshooting.xml (hunk 1 above)
Comment (tbazant) [reply xml/admin_ceph_troubleshooting.xml]: done, inserted FileStore 2 times

File: xml/admin_ceph_troubleshooting.xml (hunk 2 above)
Comment (tbazant) [reply xml/admin_ceph_troubleshooting.xml]: done

**Review by trociny — APPROVED** [review]
:+1:

File: xml/admin_ceph_troubleshooting.xml [hunk 3]
```diff
@@ -518,4 +518,35 @@ keep_jobs: 1</screen>
    </procedure>
   </tip>
  </sect1>
+ <sect1 xml:id="trouble-osd-panic">
+  <title>OSD Panic Occurs when Media Error Happens during &filestore; Directory Split</title>
+
+  <para>
+   When directory split error is generated on &filestore; OSDs, the
+   corresponding OSD is designed to terminate. It is then easier to detect the
+   problem and avoid introducing inconsistencies. If the OSD terminates
```
Comment (asettle) [inline xml/admin_ceph_troubleshooting.xml:527]:
```suggestion
   problem and avoids introducing inconsistencies. If the OSD terminates

File: xml/admin_ceph_troubleshooting.xml [hunk 4]
```diff
@@ -518,4 +518,35 @@ keep_jobs: 1</screen>
...
+   frequently, &systemd; may disable it permanently, depending on the &systemd;
+   configuration. After the OSD is disabled, it will be set out and process of
```
Comment (asettle) [inline xml/admin_ceph_troubleshooting.xml:529]:
```suggestion
    frequently, depending on the &systemd; configuration &systemd; may disable it permanently,.

**Review by asettle — APPROVED** [review]

File: xml/admin_ceph_troubleshooting.xml [hunk 5]
```diff
@@ -518,4 +518,35 @@ keep_jobs: 1</screen>
    </procedure>
   </tip>
  </sect1>
+ <sect1 xml:id="trouble-osd-panic">
+  <title>OSD Panic Occurs when Media Error Happens during &filestore; Directory Split</title>
+
+  <para>
+   When directory split error is generated on &filestore; OSDs, the
```
Comment (lproven) [inline xml/admin_ceph_troubleshooting.xml:525]: This is missing an article. Either:

When *a* directory split error is generated on *a* &filestore; OSD, the...

... or...

When directory split errors are generated on &filestore; OSDs, the...

I slightly prefer the second, as the sentence continues that _the corresponding_ OSD, implying that the one that threw the error terminates. If the sentence is in the singular, then there is no group for there to be a "corresponding" one of.

Comment (asettle) [reply xml/admin_ceph_troubleshooting.xml:525]: It should be:

When a directory split error is generated on a &filestore; OSD, the...

File: xml/admin_ceph_troubleshooting.xml [hunk 6]
```diff
@@ -518,4 +518,35 @@ keep_jobs: 1</screen>
...
+   corresponding OSD is designed to terminate. It is then easier to detect the
+   problem and avoids introducing inconsistencies. If the OSD terminates
+   frequently, &systemd; may disable it permanently depending on the &systemd;
+   configuration. After the OSD is disabled, it will be set out and process of
+   data migration will be started.
+  </para>
+
+  <para>
+   You should notice the OSD termination by regularly running the <command>ceph
+   status</command> command and perform necessary steps to investigate the
```
Comment (lproven) [inline xml/admin_ceph_troubleshooting.xml:535]: _the_ necessary steps

(if it is unambiguous what the steps are)

or

_any_ necessary steps

(if there is a choice of possible steps)

File: xml/admin_ceph_troubleshooting.xml (hunk 5 above)
Comment (tbazant) [reply xml/admin_ceph_troubleshooting.xml:525]: fixed in c1581590

**Pull request merged**